src/validate.ts

```diff
@@ -224,127 +271,116 @@ export function validate(
-        node.expression.expression.text === "Promise"
-      ) {
-        return [
-          newError(
-            node,
-            ErrorCodes.Unsupported_Use_of_Promises,
-            "Appsync does not support concurrent integration invocation or methods on the `Promise` api."
-          ),
-        ];
-      } else if (checker.isIntegrationNode(node.expression)) {
-        /**
-         * Used by AppSync to determine of a root statement can be statically analyzed.
-         */
-        const isControlFlowStatement = anyOf(
-          ts.isEmptyStatement,
-          ts.isIfStatement,
-          ts.isDoStatement,
-          ts.isWhileStatement,
-          ts.isForStatement,
-          ts.isForInStatement,
-          ts.isForOfStatement,
-          ts.isSwitchStatement,
-          ts.isLabeledStatement,
-          ts.isTryStatement
-        );
-
-        /**
-         * If this is an integration node, app sync does not support integrations outside of the main function body.
-         *
-         * We check to see if the root statement (the one in the function body), is a conditional statement, an example of support statements:
-         * variable - const v = func()
-         * expression - await func()
-         * return - return await func()
-         *
-         * Also ensure that we are not inside an inline conditional statement.
-         *
-         * const v = x ? await func() : await func2()
-         */
+    function validateAppsync(
+      node: ts.Node,
+      rootStatement?: ts.Statement
+    ): ts.Diagnostic[] {
+      if (ts.isCallExpression(node)) {
         if (
-          rootStatement &&
-          (isControlFlowStatement(rootStatement) ||
-            findParent(
-              node.expression,
-              ts.isConditionalExpression,
-              rootStatement
-            ))
+          ts.isPropertyAccessExpression(node.expression) &&
+          ts.isIdentifier(node.expression.expression) &&
+          node.expression.expression.text === "Promise"
         ) {
           return [
             newError(
               node,
-              ErrorCodes.Appsync_Integration_invocations_must_be_unidirectional_and_defined_statically
+              ErrorCodes.Unsupported_Use_of_Promises,
+              "Appsync does not support concurrent integration invocation or methods on the `Promise` api."
             ),
           ];
+        } else if (checker.isIntegrationNode(node.expression)) {
+          /**
+           * Used by AppSync to determine of a root statement can be statically analyzed.
+           */
+          const isControlFlowStatement = anyOf(
+            ts.isEmptyStatement,
+            ts.isIfStatement,
+            ts.isDoStatement,
+            ts.isWhileStatement,
+            ts.isForStatement,
+            ts.isForInStatement,
+            ts.isForOfStatement,
+            ts.isSwitchStatement,
+            ts.isLabeledStatement,
+            ts.isTryStatement
+          );
```

Comment:
Declare this at the module scope?

Reply:
just copied from compile, but yeah, will do.